### ISSUE TITLE
Could thin pool graceful cleanup thread be instantiated **not** on module import?

### ISSUE THREAD

**Issue by stephen-hansen** [oracle/python-oracledb#426]
Hi,

I'm looking at the following lines in python-oracledb:
https://github.com/oracle/python-oracledb/blob/d7003f2f848c877cd8632c8c391b4eb3bfaae068/src/oracledb/impl/thin/pool.pyx#L987-L995

When `pool.pyx` gets imported (via `thin_impl.pyx`, and then in general `__init__.py`, I think), those lines start a background thread that blocks until the main thread is finished, and then cleans up everything in pools_to_close. Following the import chain above I believe this thread is created and starts running on import of python-oracledb itself, even if pools_to_close is never populated.

This is not ideal to me, I don't expect module imports to necessarily spawn threads, or cause other side effects for that matter, at import time. Just as an example, I was working on a process earlier where I want to block SIGUSR1 in every thread in my process. Even though I was setting my main thread to block SIGUSR1, I was doing this in my main() method, and the thread spawned by python-oracledb was created at import prior to my main method even running, so it was not blocking the signals that I had intended it to. For now I am getting around this by setting blocked signals prior to import.

I'm not sure if a separate thread *is* the best way to clean up the pools, but at least one immediate idea I had here to improve the code here was to only create the cleanup thread the **first** time either something is added to the pools_to_close, or you do it at either `__init__` or `__new__` time via e.g. a class singleton, so that the first time a ThinPoolImpl is constructed the cleanup thread is started as a property of the class. The key point here being that I would greatly prefer if the cleanup thread was created closer to usage, when needed. I'm open to other ideas though.

**Comment by cjbj** [comment]
This issue rings a bell but I can't immediately find the discussion

**Comment by anthony-tuininga** [comment]
Your request has merit. I will look into this and get back to you if I have any questions!

**Comment by anthony-tuininga** [comment]
I have pushed a patch that adds this support and have initated a [build](https://github.com/oracle/python-oracledb/actions/runs/12378404871) from which you can download pre-built development wheels once it completes. You can also [build from source](https://python-oracledb.readthedocs.io/en/latest/user_guide/installation.html#installing-from-source-code) if you prefer. If you can test your scenario and confirm the patch works as expected, that would be appreciated!

**Comment by anthony-tuininga** [comment]
Yes, I knew about atexit. See the comment in the [newly committed code](https://github.com/oracle/python-oracledb/commit/a4258c6dea87ad2b87e3afe3c4dc12ee32247c8a#diff-8c53bd091416ff9411845809487258ba48a7e5a3fa8815c599293be58dc75b9bR1011-R1017) about why that wasn't an option!

**Comment by anthony-tuininga** [comment]
This has been included in python-oracledb 3.0.0 which was just released.
